vector-stores/spring-ai-elasticsearch-store/src/main/java/org/springframework/ai/vectorstore/elasticsearch/ElasticsearchVectorStore.java

```diff
@@ -188,11 +189,12 @@ public void doAdd(List<Document> documents) {
 		List<float[]> embeddings = this.embeddingModel.embed(documents, EmbeddingOptionsBuilder.builder().build(),
 				this.batchingStrategy);
 
-		for (Document document : documents) {
-			ElasticSearchDocument doc = new ElasticSearchDocument(document.getId(), document.getText(),
-					document.getMetadata(), embeddings.get(documents.indexOf(document)));
-			bulkRequestBuilder.operations(
-					op -> op.index(idx -> idx.index(this.options.getIndexName()).id(document.getId()).document(doc)));
+		for (int i = 0; i < embeddings.size(); i++) {
```

Comment:
With this change, you can delete ElasticSearchDocument now

Reply:
I deleted the `ElasticSearchDocument` record. Thanks for your feedback!